apps/system/utils/kdbg_stackmonitor.c

```diff
@@ -134,50 +139,84 @@ static void stkmon_inactive_check(void)
 	}
 }
 
-static void stkmon_active_check(struct tcb_s *tcb, void *arg)
+static void stkmon_print_active_values(char *buf)
 {
-	if (tcb->pid == 0) {
-		tcb->adj_stack_size = CONFIG_IDLETHREAD_STACKSIZE;
-		tcb->stack_alloc_ptr = (void *)(g_idle_topstack - CONFIG_IDLETHREAD_STACKSIZE);
+	int i;
+	stat_data stat_info[PROC_STAT_MAX];
+
+	stat_info[0] = strtok(buf, " ");
+
+	for (i = 1; i < PROC_STAT_MAX; i++) {
+		stat_info[i] = strtok(NULL, " ");
 	}
+
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	printf("%5d | %8s | %8d | %10d | %10d | %7lld | ", tcb->pid, "ACTIVE", tcb->adj_stack_size, up_check_tcbstack(tcb), tcb->peak_alloc_size, (uint64_t)((clock_t)clock()));
+	printf("%5s | %8s | %8s | %10s | %10s | %7lld | ", stat_info[PROC_STAT_PID], "ACTIVE", stat_info[PROC_STAT_TOTALSTACK], stat_info[PROC_STAT_PEAKSTACK], stat_info[PROC_STAT_PEAKHEAP], (uint64_t)((clock_t)clock()));
 #else
-	printf("%5d | %8s | %8d | %10d | %10lld | ", tcb->pid, "ACTIVE", tcb->adj_stack_size, up_check_tcbstack(tcb), (uint64_t)(clock()));
+	printf("%5s | %8s | %8s | %10s | %10s | ", stat_info[PROC_STAT_PID], "ACTIVE", stat_info[PROC_STAT_TOTALSTACK], stat_info[PROC_STAT_PEAKSTACK], (uint64_t)((clock_t)clock()));
 #endif
 #if (CONFIG_TASK_NAME_SIZE > 0)
-	printf("%s\n", tcb->name);
+	printf("%s\n", stat_info[PROC_STAT_NAME]);
 #else
 	printf("\n");
 #endif
 }
 
+static int stkmon_read_pid(FAR struct dirent *entryp, FAR void *arg)
+{
+	int ret;
+	char *filepath;
+	char buf[STKMON_BUFLEN];
+
+	asprintf(&filepath, "%s/%s/%s", PROCFS_MOUNT_POINT, entryp->d_name, "stat");
+	ret = kdbg_readfile(filepath, buf, STKMON_BUFLEN, stkmon_print_active_values);
+	free(filepath);
+	if (ret < 0) {
+		printf("Failed to read %s\n", filepath);
+		return ERROR;
+	}
+
+	return OK;
+}
+
 static void *stackmonitor_daemon(void *arg)
 {
+	int ret;
+
+	/* Mount Procfs to use */
+	ret = mount(NULL, PROCFS_MOUNT_POINT, PROCFS_FSTYPE, 0, NULL);
```

Comment:
I think you'd better move mount to kdbg_stackmonitor().
This is a daemon which runs periodically, but there is not needed to mount for every time.

Reply:
I think it is right because mount is out of while loop.